### DISCUSSION TITLE
If the player type "/Command 1", I want to print out the message, what should I do?

### DISCUSSION THREAD

**Discussion by minhos1106** [Minestom/Minestom#1042] [Help]
If the player type "/Command 1", I want to print out the message, what should I do?

**Comment by kiip1** [comment]
Use CommandContext#getInput in the executor to get the typed command

Example:
```java
// In your command constructor
setDefaultExecutor(((sender, context) -> {
    sender.sendMessage(context.getInput());
}));
```

**Reply by minhos1106** [reply]
If the number is 1, I want to print out a message, what should I do?

**Reply by GreatWyrm** [reply]
What is the message that you want to print out, and where do you want to print it out to? Other players? The console?

**Reply by minhos1106** [reply]
> What is the message that you want to print out, and where do you want to print it out to? Other players? The console?

thx :)

**Comment by MrGazdag** [comment]
The question is a bit vague, so heres a list of possible solutions:
- If you want to create a command, then create your own class, which extends [`Command`](https://github.com/Minestom/Minestom/blob/master/src/main/java/net/minestom/server/command/builder/Command.java), and register it in the [`CommandManager`](https://github.com/Minestom/Minestom/blob/master/src/main/java/net/minestom/server/command/CommandManager.java) using `register(Command)`:
```java
// If your custom command class is named MyCommand:
MinecraftServer.getCommandManager().register(new MyCommand());
```
- If you would like to run code when a player types something in chat, listen to [`PlayerChatEvent`](https://github.com/Minestom/Minestom/blob/master/src/main/java/net/minestom/server/event/player/PlayerChatEvent.java)s. From there, you can get the message like this:
```java
MinecraftServer.getGlobalEventHandler().addListener(PlayerChatEvent.class, e->{
    String message = e.getMessage();
    // Do something with message
});
```
- If you want to query specific arguments in a custom command, you can use `CommandContext#get(Argument)` or `CommandContext#get(String)`, for example:
```java
public class DuckCommand extends Command {
    public DuckCommand() {
        super("duck"); // The command name
        
        // Arguments
        ArgumentLiteral call = ArgumentType.Literal("call"); // The player needs to type "call" exactly as it is
        ArgumentString duckName = ArgumentType.String("duckName"); // The player can type whatever they want (there are some rules, like no spaces allowed, etc)
        
        // Register the command
        addSyntax((sender, context) -> {
            String chosenDuck = context.get(duckName);
            sender.sendMessage("Come here, " + chosenDuck + "!");
        }, call, duckName); // This will register a command with the syntax /duck call <duckName>
    }
}
```
- If you would like to get the whole command that the player typed in a command, use `CommandContext#getInput()`:
```java
addSyntax((sender, context) -> {
    String wholeMessage = context.getInput();
    // Do something with wholeMessage
});
```

There's more information on commands on [the wiki](https://wiki.minestom.net/feature/commands).
As @kiip1 said though, next time use [Github Discussions](https://github.com/Minestom/Minestom/discussions/new?category=help), issues are usually for Minestom bugs.

**Reply by minhos1106** [reply]
If the number is 1, I want to print out a message, what should I do?

**Reply by MrGazdag** [reply]
You can get the value of the argument as you can see in the third example:
```java
// Register the command
addSyntax((sender, context) -> {
    String chosenDuck = context.get(duckName);
    sender.sendMessage("Come here, " + chosenDuck + "!");
}, call, duckName); // This will register a command with the syntax /duck call <duckName>
```
Here i am getting the value of the `duckName` argument, by using `context.get(duckName)`. You can get your argument's value in the same way. After that, just check if it is 1, then do what you want to do.

**Reply by minhos1106** [reply]
thx :)

**Reply by MrGazdag** [reply]
No worries, also, here on Github Discussions, you can select an answer and mark it as a solution to mark this question as solved. This indicates to people that want to help, that this question does not need further attention.
